sandy/core/cov.py

```diff
@@ -306,41 +306,13 @@ def eig(self, sparse=False, sort=True):
                     0            1
         0 7.07107e-01 -7.07107e-01
         1 7.07107e-01  7.07107e-01
-
-        >>> sandy.CategoryCov([[1, 0.4],[0.4, 1]]).eig(sparse=True)[0]
-        0   1.40000e+00
-        1   6.00000e-01
-        Name: eigenvalues, dtype: float64
-
-        Extract eigenfunctions of covariance matrix.
-        >>> sandy.CategoryCov([[1, 0.4],[0.4, 1]]).eig(sparse=True)[1]
-                    0            1
-        0 7.07107e-01 -7.07107e-01
-        1 7.07107e-01  7.07107e-01
-
-        >>> sandy.CategoryCov(np.eye(13)).eig(sparse=True)[0]
-        0   1.00000e+00
-        1   1.00000e+00
-        2   1.00000e+00
-        3   1.00000e+00
-        4   1.00000e+00
-        5   1.00000e+00
-        Name: eigenvalues, dtype: float64
-
-        Extract eigenfunctions of covariance matrix.
-        >>> sandy.CategoryCov(np.eye(13)).eig(sparse=True)[1].values.shape
-        (6, 6)
-        """
-        if sparse:
-            cov = self.to_sparse()
-            warnings.filterwarnings("ignore")
-            try:
-                E, V = spsl.eigs(cov)
-            except TypeError or ValueError:
-                E, V = scipy.linalg.eig(cov.toarray())
-        else:
-            cov = self.data
-            E, V = scipy.linalg.eig(cov)
+        """
+        cov = self.to_sparse()
+        warnings.filterwarnings("ignore")
+        try:
+            E, V = spsl.eigs(cov)
+        except TypeError or ValueError:

```

Comment:
Is not always posible to use sparse.linalg module. Some times gives TypeError or ValueError

Reply:
I don't agree...not according to this https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.linalg.eigs.html

And even if it were true  then why `scipy.linalg.eig` would work?

Reply:
I don't agree...not according to https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.linalg.eigs.html

And even if it were true why scipy.linalg.eig would work?

Reply:
The problem with this method is that a very large sparcity is needed for it to work. For example, the test arrays give the following error: TypeError: Cannot use scipy.linalg.eig for sparse A with k >= N - 1. Use scipy.linalg.eig(A.toarray()) or reduce k.

That's why I put a try, so that small arrays are done with scipy.linalg and large arrays are done with scipy.sparse.linalg.